### PR TITLE
feat: [Operator] Retrieve autogenerated Keycloak admin user credentials

### PR DESCRIPTION
**Pull request by mmorhun**
Signed-off-by: Mykola Morhun <mmorhun@redhat.com>

### What does this PR do?
Retrieves and prints Keycloak admin-user login and password after Che installation finishes.
This implementation is done for `operator` and `olm` installers.
The same functionality has been implemented for `helm` installer before https://github.com/che-incubator/chectl/pull/686

`chectl` output:
```
    ...
    ✔ Eclipse Che pod bootstrap
      ✔ scheduling...done.
      ✔ downloading images...done.
      ✔ starting...done.
    ✔ Retrieving Eclipse Che server URL... https://che-che.192.168.99.212.nip.io
    ✔ Eclipse Che status check
  ✔ Retrieving Keycloak admin credentials... admin:2h9nRBK7SCPr
  ✔ Retrieving Che self-signed CA certificate... is exported to /home/username/cheCA.crt
  ✔ Show important messages
    ✔ Eclipse Che admin credentials are: "admin:admin". You will be asked to change default Che admin password on the first login.
    ✔ Autogenerated Keycloak credentials are: "admin:2h9nRBK7SCPr"
    ✔ ❗[MANUAL ACTION REQUIRED] Please add Che self-signed CA certificate into your browser: /home/mmorhun/cheCA.crt.
   Documentaton how to add a CA certificate into a browser: https://www.eclipse.org/che/docs/che-7/installing-che-in-tls-mode-with-self-signed-certificates/#using-che-with-tls_installing-che
-in-tls-mode-with-self-signed-certificates
Command server:start has completed successfully.
```

### What issues does this PR fix or reference?
https://github.com/eclipse/che/issues/14082

### Depends on
https://github.com/eclipse/che-operator/pull/240

### CODE REVIEW

File: src/api/che.ts [hunk 1]
```diff
@@ -139,6 +140,34 @@ export class CheHelper {
...
+   * Retreives Keycloak admin user credentials.
+   * Works only with installers which use Che CR (operator, olm).
+   * Returns credentials as an array of two values: [login, password]
+   * In case of an error an array with undefined values will be returned.
+   */
+  async retrieveKeycloakAdminCredentials(cheNamespace: string): Promise<string[]> {
+    let adminUsername
+    let adminPassword
+
+    const cheCluster = await this.kube.getCheCluster('eclipse-che', cheNamespace)
```
Comment (tolusha) [inline src/api/che.ts:153]: Use the constant
https://github.com/che-incubator/chectl/blob/master/src/constants.ts#L25

File: src/api/che.ts [hunk 2]
```diff
@@ -139,6 +140,34 @@ export class CheHelper {
...
+    const keycloakCredentialsSecretName = cheCluster.spec.auth.identityProviderSecret
+    if (keycloakCredentialsSecretName) {
+      // Keycloak credentials are sotored in secret
```
Comment (tolusha) [inline src/api/che.ts:156]: stored

File: src/api/che.ts [hunk 3]
```diff
@@ -139,6 +140,34 @@ export class CheHelper {
...
+      const keycloakCredentialsSecret = await this.kube.getSecret(keycloakCredentialsSecretName, cheNamespace)
+      if (keycloakCredentialsSecret && keycloakCredentialsSecret.data) {
+        adminUsername = base64Decode(keycloakCredentialsSecret.data.user)
+        adminPassword = base64Decode(keycloakCredentialsSecret.data.password)
+      }
+    } else {
+      // Keycloak credentials are sotored in Che custom resource
```
Comment (tolusha) [inline src/api/che.ts:163]: stored

File: src/tasks/installers/common-tasks.ts [hunk 4]
```diff
@@ -154,3 +154,21 @@ export function getMessageImportCaCertIntoBrowser(caCertFileLocation: string): s
                   `Documentaton how to add a CA certificate into a browser: ${DOCS_LINK_IMPORT_CA_CERT_INTO_BROWSER}`
   return message
 }
+
+export function getRetrieveKeycloakCredentialsTask(flags: any): ListrTask {
+  return {
+    title: 'Retrieving Keycloak admin credentials',
+    enabled: () => flags.multiuser && (flags.installer !== 'operator' || flags.installer !== 'olm'),
```
Comment (tolusha) [inline src/tasks/installers/common-tasks.ts:161]: let's skip if 
!spec.auth.ExternalIdentityProvider

**Comment by tolusha** [comment]
Update yarn.lock to rely on the latest che-operator changes.

File: src/tasks/installers/common-tasks.ts (hunk 4 above)
Comment (mmorhun) [reply src/tasks/installers/common-tasks.ts:161]: Good catch

**Comment by mmorhun** [comment]
Yes, I do remember. But thank you for reminding me.

**Pull request merged**